src/domain/challenge/challenge/challenge.resolver.mutations.ts

```diff
@@ -90,4 +94,18 @@ export class ChallengeResolverMutations {
   ): Promise<IChallenge> {
     return await this.challengeService.removeChallengeLead(removeData);
   }
+
+  @Roles(AuthorizationRoles.EcoverseAdmins, AuthorizationRoles.GlobalAdmins)
+  @UseGuards(GqlAuthGuard)
+  @Mutation(() => Challenge, {
+    description: 'Trigger an event on the Challenge.',
+  })
+  async eventOnChallenge(
+    @Args('challengeEventData')
+    challengeEventData: ChallengeLifecycleEventInput
+  ): Promise<IChallenge> {
+    return await this.challengeLifecycleOptionsProvider.eventOnChallenge(
```

Comment:
It might make sense to move eventOnChallenge (as a pattern) on the level of the root service (challengeService in this case).

Reply:
Agreed. Cleaner - then the options provider is only for the options.